src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1187,21 +1179,60 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     const common = this.fallback
       ? this.fallback.getCommonForBlockNumber(
           this.common,
-          BigInt(newBlock.header.number.toString())
+          BigInt(block.header.number.toString())
         )
       : this.common;
 
     const vm = await VM.create({
-      state: trie,
       activatePrecompiles: false,
       common,
       allowUnlimitedContractSize: this.vm.allowUnlimitedContractSize,
       blockchain,
       stateManager: this.fallback
         ? new ForkStateManager({ common, trie: trie as ForkTrie })
-        : new DefaultStateManager({ common, trie: trie })
+        : new GanacheStateManager({ common, trie: trie })
     });
 
+    // Don't even let the vm try to flush the block's _cache to the stateTrie.
+    // When forking some of the data that the traced function may request will
+    // exist only on the main chain. Because we pretty much lie to the VM by
+    // telling it we DO have data in our Trie, when we really don't, it gets
+    // lost during the commit phase when it traverses the "borrowed" datum's
+    // trie (as it may not have a valid root). Because this is a trace, and we
+    // don't need to commit the data, duck punching the `flush` method (the
+    // simplest method I could find) is fine.
+    // Remove this and you may see the infamous
+    // `Uncaught TypeError: Cannot read property 'pop' of undefined` error!
+    (vm.stateManager as any)._cache.flush = () => {};
+
+    // Process the block without committing the data.
+    await vm.stateManager.checkpoint();
+
+    for (let i = 0; i < transactionIndex; i++) {
+      const tx = block.transactions[i] as any;
+      const transactionEventContext = {};
+      this.emit("ganache:vm:tx:before", {
+        context: transactionEventContext
+      });
+      await vm.runTx({ tx, block: block as any });
+      this.emit("ganache:vm:tx:after", {
+        context: transactionEventContext
+      });
+    }
```

Comment:
This is pretty much the same as before, but now we don't have weird branching just for `storageRangeAt`